src/main/java/org/pethotel/HeavenForPets/service/impl/RoomServiceImpl.java

```diff
@@ -194,7 +199,7 @@ public int getNumberOfRoomsFromQuery() {
             throw new TemperatureWrongRangeException();
         }
 
-        List<RoomEntity> roomEntityList = roomRepository.findAllPlantRooms();
+        List<RoomEntity> roomEntityList = getAllPlantRooms();
```

Comment:
You can create in Repository method called: "getAllPlantRoomsForTemperature(min, max)" then the DB will prepare collections of Rooms with temperature and then the checking in loop is not neeaded:

 if (petEntity.getMinTemperature() <= roomEntity.getTemperature()
&& petEntity.getMaxTemperature() >= roomEntity.getTemperature()){

Reply:
Will be added in future issue